### ISSUE TITLE
We need a way to identify when a device is running more than one Code Engine instance

### ISSUE THREAD

**Issue by marusshi** [karthikp-ubc/ThingsJS#20]
The Scheduler algorithm takes into account the current available memory of each live device.
Currently we expect that each device only runs one Code Engine instance, and the Code Engine instance will broadcast the device's available memory.

However if one device runs _n_ Code Engine instances, the Scheduler will think there is _n_ times the available memory. 

Perhaps each Code Engine can also provide in its meta what device it belongs to (e.g. via MAC address).

**Comment by juliengs** [comment]
Good point! I think we discussed it briefly at the last meeting -- my proposal (in the meanwhile) would be to divide the total system available by _n_ if _n_ code engines are running on the same device (would there be a way for a code engine to detect if other code engines are running on the same device?).

I think this is an OK solution, as logically, in a real deployment, I don't see why we would need to run more than one code engines per device. We do this pretty much only for testing.

**Comment by jungkumseok** [comment]
Yes I think we definitely need the ability to determine how many CodeEngines are running on a single device, but I don't know if we should assume or impose restriction that there will be only one CodeEngine running per device. Although in practice a single user would have no reason to run multiple CodeEngines, many cloud/servers are time-shared and there could be multiple users ssh'ing into the same machine and running a CodeEngine per logged-in user. In such cases there will be multiple CodeEngines running on the same machine and the user would not have control over other users taking up resources. So I think we should allow multiple CodeEngines on the same device as long as we can detect how many CodeEngines are sharing resources.
